static/src/javascripts/projects/commercial/modules/creatives/template.js

```diff
@@ -97,10 +97,12 @@ define([
 
                 var creativeHtml = template(creativeTpl, this.params);
 
-                resolve(fastdom.write(function () {
+                fastdom.write(function () {
                     this.adSlot.insertAdjacentHTML('beforeend', creativeHtml);
-                    return true;
-                }, this));
+                }, this)
+                .then(function () {
+                    resolve(true);
+                });
```

Comment:
Does the `true` passed to resolve matter?  Would `.then(resolve)` work in this case?

Also: @gtrufitt this looks familiar!